### DISCUSSION TITLE
combining values - adding useWheel to useDrag

### DISCUSSION THREAD

**Discussion by carmandale** [pmndrs/use-gesture#258] [Q&A]
I finally made a decent useDrag work with R3F. ...very proud...it works well on desktop and mobile. Now I want to add useWheel so that your mouse can drag it or your wheel can move it. I know that I can change useDrag to useGesture and add useWheel, but I am not sure how I combine the values. ultimately it would be combining the wheelY scroll in a smooth, seamless way. Can someone help me out? or point me in the right direction?

`function DraggableNeighborhood() {
  const ref = useRef();
  const { viewport } = useThree();
  const [spring, set] = useSpring(() => ({
    position: [0, 0, 3],
    config: config.molasses,
  }));

  const bind = useDrag(
    ({ offset: [x, y] }) => {
      const aspect = viewport().factor;
      set({
        position: [
          clamp(x / aspect, -5.7, 7.0),
          clamp(-y / aspect, -0.1, 0.1),
          3,
        ],
      });
    },
    { pointerEvents: true }
  );

  return (
    <a.mesh {...spring} {...bind()} ref={ref}>
      <Model />
    </a.mesh>
  );
}`

**Comment by dbismut** [comment]
Please start with formatting your code properly. Also it looks like you're using an outdated version of this lib.
